From 443b5c0317a81d95a0dffde9f566d47a106b962b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 10 Feb 2020 15:54:35 +0200
Subject: [PATCH] [SW-0] pll: introduce weak hl_pll_get_cfg() api

Introduce the weak variant of the hl_pll_get_cfg() function, so there
will be no need to implement it for each H/W variant (e.g. PLDM, ONIX,
etc.).
Remove the anymore unneeded implementations.

Change-Id: I00c1903ecf3b42b464276e3741ac6648d598ebe8
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index b4006c1..a82c2c6 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -217,6 +217,9 @@
 __attribute__((weak))
 bool hl_pll_is_configurable(struct hl_pll_cfg *cfg) { return true; }
 
+__attribute__((weak))
+int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg) { return 0; }
+
 /**
  * hl_pll_init - Initialize PLL registers
  */
diff --git a/gaudi/onix.c b/gaudi/onix.c
index cd7e74e..c7abdd0 100644
--- a/gaudi/onix.c
+++ b/gaudi/onix.c
@@ -13,11 +13,6 @@
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	return 0;
-}
-
 u32 hl_skip_pcie_init(void)
 {
 	return 1;
diff --git a/gaudi/pldm.c b/gaudi/pldm.c
index 35018b4..26d30f6 100644
--- a/gaudi/pldm.c
+++ b/gaudi/pldm.c
@@ -15,11 +15,6 @@
 #define UART_CLOCK_FREQUENCY	(SYS_CLOCK_FREQUENCY / 5)
 #define PLDM_UART_BAUDRATE	1200
 
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	return 0;
-}
-
 u32 hl_skip_pcie_init(void)
 {
 	return 1;
diff --git a/goya/onix.c b/goya/onix.c
index c53ea29..ce73ff8 100644
--- a/goya/onix.c
+++ b/goya/onix.c
@@ -13,11 +13,6 @@
 
 #define SYS_CLOCK_FREQUENCY	50000000ULL
 
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	return 0;
-}
-
 u32 hl_skip_pcie_init(void)
 {
 	return 1;
diff --git a/goya2/onix.c b/goya2/onix.c
index cd7e74e..c7abdd0 100644
--- a/goya2/onix.c
+++ b/goya2/onix.c
@@ -13,11 +13,6 @@
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	return 0;
-}
-
 u32 hl_skip_pcie_init(void)
 {
 	return 1;
diff --git a/goya2/pldm.c b/goya2/pldm.c
index 35018b4..26d30f6 100644
--- a/goya2/pldm.c
+++ b/goya2/pldm.c
@@ -15,11 +15,6 @@
 #define UART_CLOCK_FREQUENCY	(SYS_CLOCK_FREQUENCY / 5)
 #define PLDM_UART_BAUDRATE	1200
 
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	return 0;
-}
-
 u32 hl_skip_pcie_init(void)
 {
 	return 1;
diff --git a/goya2_arc/onix.c b/goya2_arc/onix.c
index 8ad210f..ef00ec3 100644
--- a/goya2_arc/onix.c
+++ b/goya2_arc/onix.c
@@ -15,11 +15,6 @@
 #define UART_CLOCK_FREQUENCY	(SYS_CLOCK_FREQUENCY / 5)
 #define PLDM_UART_BAUDRATE	57600
 
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	return 0;
-}
-
 u32 hl_skip_pcie_init(void)
 {
 	return 1;
diff --git a/goya2_arc/pldm.c b/goya2_arc/pldm.c
index 3f6b3d7..31906a7 100644
--- a/goya2_arc/pldm.c
+++ b/goya2_arc/pldm.c
@@ -15,11 +15,6 @@
 #define UART_CLOCK_FREQUENCY	(SYS_CLOCK_FREQUENCY / 5)
 #define PLDM_UART_BAUDRATE	1200
 
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	return 0;
-}
-
 u32 hl_skip_pcie_init(void)
 {
 	return 0;
